src/analyses/goto_check.cpp

```diff
@@ -1031,27 +1031,16 @@ goto_checkt::address_check(const exprt &address, const exprt &size)
         not_exprt(dead_object(address, ns)), "dead object"));
     }
 
-    if(flags.is_unknown() || flags.is_dynamic_heap())
-    {
-      const or_exprt dynamic_bounds_violation(
-        dynamic_object_lower_bound(address, ns, nil_exprt()),
-        dynamic_object_upper_bound(address, ns, size));
```

Comment:
To be perfectly clean, shouldn't all the code implementing `dynamic_object_*_bound` be removed as well?